torch/csrc/jit/script/compiler.cpp

```diff
@@ -1422,6 +1422,50 @@ struct to_ir {
     emitLoopCommon(range, body, assigner, {}, max_trip_count_val);
   }
 
+  void emitForInTensorLoop(const For& stmt, Value* tensorArg) {
```

Comment:
The `emitForIn...` methods all seem very similar, maybe we can refactor the common stuff into a single method.

Reply:
Okay I gave that a try. It doesn't save us much: https://phabricator.internmc.facebook.com/P62623620$1374 .Also, it makes the conceptual continuity of those functions worse.